hackathon_site/dashboard/frontend/src/components/general/SideSheetR/SideSheetR.js

```diff
@@ -0,0 +1,165 @@
+            </div>
+            <div className={styles.bodyinfo}>
+                <Typography variant="h3">Datasheet</Typography>
+                <Link
+                    href={detail.datasheet}
+                    rel="noopener"
+                    color="inherit"
+                    underline="none"
+                    target="_blank"
+                    style={{ display: "flex" }}
+                >
+                    <LaunchIcon></LaunchIcon>
+                    <Typography>Link</Typography>
+                </Link>
+            </div>
+            <div className={styles.bodyinfo}>
+                <Typography variant="h3">Notes</Typography>
+                {detail.notes.map((note, i) => (
+                    <Typography key={i}>{note}</Typography>
+                ))}
+            </div>
+            <div className={styles.bodyinfo}>
+                <Typography variant="h3">Constraints</Typography>
+                {detail.constraints.map((constraint, i) => (
+                    <Typography key={i}>{constraint}</Typography>
+                ))}
+            </div>
+        </div>
+    );
+};
+
+const HeaderSheet = ({ detail }) => {
+    return (
+        <div className={styles.headersheet}>
+            <div>
+                <div className={styles.title}>
+                    <CircleType
+                        style={{ color: detail.type, margin: "0 5px" }}
+                    ></CircleType>
+                    <Typography variant="h2">{detail.name}</Typography>
+                </div>
+                <Typography className={styles.available}>
+                    {detail.available} OF {detail.total} IN STOCK
+                </Typography>
+                <div className={styles.tags}>
+                    <Typography variant="h3">Tags</Typography>
+                    <Typography>{detail.tags}</Typography>
+                </div>
+            </div>
+            <img src={detail.img} alt="product" />
+        </div>
+    );
+};
+
+const SideSheetR = ({ detail, cart }) => {
+    const [toggle, setToggle] = React.useState(false);
+
+    const handleChange = (change) => {
+        setToggle(change);
+    };
+
+    return (
+        <Grid>
+            <div
+                onClick={() => {
+                    handleChange(true);
+                }}
+            >
+                click me
+            </div>
+            <Drawer anchor="right" open={toggle}>
+                <div className={styles.topsheet}>
+                    <BackArrow
+                        className={styles.backarrow}
+                        onClick={() => {
+                            handleChange(false);
+                        }}
+                        role="close"
+                    />
+                    <Typography variant="h2" className={styles.title}>
+                        Product Overview
+                    </Typography>
+                </div>
+                <HeaderSheet detail={detail} />
```

Comment:
Rather than passing `detail` down into all of these components, I would recomment breaking out props for exactly what information you need and only passing that in. It makes it easier to trace which data is being used where.